build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -916,6 +916,23 @@ message Digest {
   int64 size_bytes = 2;
 }
 
+message Blob {
+  // The first two field types + numbers must match those in Digest.
+  // This is a hack to allow repurposing Digest fields, which we can
+  // replace with Blobs and old clients will ignore the newly added
+  // fields (which should be unused by old clients anyway).
+  //
+  // For V3, we might consider embedding a Digest inside Blob, but
+  // that would require deprecating more fields.
+  string hash = 1;
+  int64 size_bytes = 2; // This refers to the uncompressed size
+
+  reserved 3, 4; // Leave some room in case we want to extend Digest later(?)
```

Comment:
If we add more fields to Digest, then clients/servers will already update to a newer version of the .proto. In that case they will already switch from Digest to Blob.

In other words, I don't see how reserving numbers here helps us.

Reply:
You might be right. If we update Digest later, we could reserve fields in Digest to "jump over" the extra fields in Blob.

Reply:
If we update Digest later, it's completely safe to use any numbers that overlap with Blob.